scripts/run_e2e_tests.py

```diff
@@ -330,6 +333,39 @@ def run_component_template_e2e_test(ctx: Context, template_dir: str) -> bool:
     return success
 
 
+def is_app_server_alive():
+    try:
+        r = requests.get("http://localhost:3000/", timeout=3)
+        return r.status_code == requests.codes.ok
+    except:
+        return False
+
+
+def run_app_server():
+    if is_app_server_alive():
+        print("Detected React app server already running, won't spawn a new one.")
+        return
+
+    env = {
+        "BROWSER": "none",  # don't open up chrome, streamlit does this for us
+        "DISABLE_HARDSOURCE_CACHING": "true",
+        "GENERATE_SOURCEMAP": "false",
+        "INLINE_RUNTIME_CHUNK": "false",
+    }
+    command = ["yarn", "start", "--running-streamlit-e2e-test"]
+    proc = AsyncSubprocess(command, cwd=FRONTEND_DIR, env=env)
+
+    print("Starting React app server...")
+    proc.start()
+
+    print("Waiting for React app server to come online...")
+    while not is_app_server_alive():
```

Comment:
Maybe add a timeout so this doesn't wait forever if something goes wrong with the react app server startup?